pulser-core/pulser/_seq_drawer.py

```diff
@@ -594,11 +608,11 @@ def phase_str(phi: float) -> str:
             if ax_lims[i][0] < 0:
                 ax.axhline(0, **hline_kwargs)
 
-        if "interp_pts" in data[ch] and draw_interp_pts:
-            all_points = data[ch]["interp_pts"]
-            for i, qty in enumerate(("amplitude", "detuning")):
-                if qty in all_points and curves_per_ch[ch][i]:
-                    pts = np.array(all_points[qty])
-                    axes[i].scatter(pts[:, 0], pts[:, 1], color=COLORS[i])
+        if ch_data.interp_pts and draw_interp_pts:
```

Comment:
`if draw_interp_pts` would probably be enough, since `qty in ch_data.interp_pts` will always be `False` if `ch_data.interp_pts` is empty.